go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -339,13 +340,19 @@ func (c *Controller) syncServices(key string) error {
 				}
 				// Configure the NodePort in each Node Gateway Router
 				for _, gatewayRouter := range gatewayRouters {
-					lbID, err := gateway.GetGatewayLoadBalancer(gatewayRouter, svcPort.Protocol)
+					gatewayLB, err := gateway.GetGatewayLoadBalancer(gatewayRouter, svcPort.Protocol)
 					if err != nil {
 						klog.Warningf("Service Sync: Gateway router %s does not have load balancer (%v)",
 							gatewayRouter, err)
 						// TODO: why continue? should we error and requeue and retry?
 						continue
 					}
+					workerNode := strings.TrimPrefix(gatewayRouter, "GR_")
```

Comment:
if we are going to settle on this convention this should be an API in the pkg/ovn/gateway, `getWorkerFromRouter()` or something like this

Reply:
will do

Reply:
note this current code is repeated everywhere in the code base: 

```
pkg/ovn/egressgw.go:		gr := "GR_" + pod.Spec.NodeName
pkg/ovn/egressgw.go:			node := strings.TrimPrefix(gr, "GR_")
pkg/ovn/egressgw.go:			node := strings.TrimPrefix(gr, "GR_")
pkg/ovn/egressgw.go:				node := strings.TrimPrefix(gr, "GR_")
pkg/ovn/egressgw.go:	gr := "GR_" + node
pkg/ovn/egressgw.go:	gr := "GR_" + node
pkg/ovn/controller/services/utils.go:			workerNode := strings.TrimPrefix(gatewayRouter, "GR_")
pkg/ovn/controller/services/utils.go:			workerNode := strings.TrimPrefix(gatewayRouter, "GR_")
pkg/ovn/controller/services/utils.go:			workerNode := strings.TrimPrefix(gatewayRouter, "GR_")
pkg/ovn/egressip.go:					fmt.Sprintf("GR_%s", status.Node),
pkg/ovn/egressip.go:					fmt.Sprintf("GR_%s", status.Node),
pkg/ovn/master.go:			_, stderr, err := util.RunOVNNbctl("--may-exist", "lr-lb-add", "GR_"+node.Name, clusterLB)
pkg/ovn/master.go:				return fmt.Errorf("unable to add cluster LB: %s to GR_%s, stderr: %q, error: %v",
pkg/ovn/master.go:		gatewayRouter := "GR_" + node.Name
pkg/ovn/gateway.go:			workerNode := strings.TrimPrefix(gatewayRouter, "GR_")
pkg/ovn/gateway.go:			workerNode := strings.TrimPrefix(gatewayRouter, "GR_")
pkg/ovn/endpoints.go:			workerNode := strings.TrimPrefix(gateway, "GR_")
pkg/types/const.go:	GWRouterPrefix               = "GR_"
```

So I'll come up with another commit to change it everywhere.